uat/custom-components/src/main/java/com/aws/greengrass/artifacts/ShadowComponent.java

```diff
@@ -0,0 +1,221 @@
+
+        UpdateThingShadowResponse updateThingShadowResponse =
+                greengrassCoreIPCClient.updateThingShadow(updateThingShadowRequest, Optional.empty()).getResponse().get();
+
+        JsonNode receivedShadowDocumentJson = MAPPER.readTree(updateThingShadowResponse.getPayload());
+        removeTimeStamp(receivedShadowDocumentJson);
+        removeMetadata(receivedShadowDocumentJson);
+    }
+
+    void handleDeleteThingShadowOperation(String thingName, String shadowName)
+            throws ExecutionException, InterruptedException, IOException {
+        DeleteThingShadowRequest deleteThingShadowRequest = new DeleteThingShadowRequest();
+        deleteThingShadowRequest.setThingName(thingName);
+        deleteThingShadowRequest.setShadowName(shadowName);
+
+        DeleteThingShadowResponse deleteThingShadowResponse =
+                greengrassCoreIPCClient.deleteThingShadow(deleteThingShadowRequest, Optional.empty()).getResponse().get();
+    }
+
+    void handleListNamedShadowsForThingOperation(String thingName, Integer pageSize, String nextToken)
+            throws ExecutionException, InterruptedException, IOException {
+        ListNamedShadowsForThingRequest listNamedShadowsForThingRequest = new ListNamedShadowsForThingRequest();
+        listNamedShadowsForThingRequest.setThingName(thingName);
+        listNamedShadowsForThingRequest.setPageSize(pageSize);
+        listNamedShadowsForThingRequest.setNextToken(nextToken);
+
+        ListNamedShadowsForThingResponse listNamedShadowsForThingResponse =
+                greengrassCoreIPCClient.listNamedShadowsForThing(listNamedShadowsForThingRequest, Optional.empty()).getResponse().get();
+
+        List<String> resultList = listNamedShadowsForThingResponse.getResults();
+
+        Integer namedShadowCount = resultList.size();
+        LOGGER.info(String.format("Returned %s from list named shadow for thingName: %s", namedShadowCount, thingName));
+        // expected to get token if returned list is equal to pageSize
+        if (namedShadowCount.equals(pageSize)) {
+            String returnToken = listNamedShadowsForThingResponse.getNextToken();
+            if (returnToken != null && !returnToken.isEmpty()) {
+                LOGGER.info(String.format("Retrieved token %s from list named shadow for thingName: %s", returnToken, thingName));
+            } else {
+                LOGGER.info("Expected token not returned in response");
+            }
+        }
+
+        for (String namedShadowResult : resultList) {
+            LOGGER.info(String.format("Retrieved named shadow %s from list named shadow for thingName: %s", namedShadowResult, thingName));
+        }
+    }
+
+    // Setup helper function which creates four named shadows for a particular thing
+    void handleSetupForListNamedShadows(String thingName, byte[] updateDocument)
+            throws ExecutionException, InterruptedException, IOException {
+        for (String namedShadow : NAMED_SHADOWS_LIST) {
```

Comment:
We could completely remove this method if we change the method signature of `handleUpdateThingShadowOperation` to 

```java
handleUpdateThingShadowOperation(String thingName, byte[] updateDocument, String... namedShadows) {
```